### ISSUE TITLE
Non-existant vars file causes nomad run to trigger a segmentation fault

### ISSUE THREAD

**Issue by andrewgwalker** [hashicorp/nomad#10867]
### Nomad version
Output from `nomad version`

Nomad v1.0.4+ent (72adb9f0e0ca0ac142f783fd62891840bc9029ba)

### Operating system and Environment details
Red Hat Enterprise Linux Server release 7.9 (Maipo)

### Issue
Trying to run a job from the nomad command line tool produces a segmentation fault when the vars file does not exist.

### Reproduction steps
Try and run any job and pass in a var file that does not actually exist e.g

`nomad job run --var-file="non_existent_file.vars"  workers.nomad`

#### Expected Result
An error message regarding the non-existent vars file.

#### Actual Result
```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x28cdbbb]

goroutine 1 [running]:
github.com/hashicorp/nomad/jobspec2.(*jobConfig).collectInputVariableValues(0xc0002ae4d0, 0xc000102a80, 0x18, 0x18, 0xc000132f80, 0x1, 0x1, 0xc0004c3140, 0xc00085d790, 0x2, ...)
	github.com/hashicorp/nomad/jobspec2/types.variables.go:548 +0x113b
github.com/hashicorp/nomad/jobspec2.(*jobConfig).decodeBody(0xc0002ae4d0, 0x3cfe020, 0xc000873290, 0xc000132f78, 0x1, 0x1)
	github.com/hashicorp/nomad/jobspec2/types.config.go:66 +0x245
github.com/hashicorp/nomad/jobspec2.decode(0xc0002ae4d0, 0xc0002ae4d0, 0x375f715)
	github.com/hashicorp/nomad/jobspec2/parse.go:95 +0x258
github.com/hashicorp/nomad/jobspec2.ParseWithConfig(0xc0001f6580, 0xc0001f6580, 0x18, 0xc000132a50)
	github.com/hashicorp/nomad/jobspec2/parse.go:44 +0xa8
github.com/hashicorp/nomad/command.(*JobGetter).ApiJobWithArgs(0xc00001f1c0, 0x7ffec9c3d6be, 0xd, 0x0, 0x0, 0x0, 0xc0001a9420, 0x1, 0x1, 0x0, ...)
	github.com/hashicorp/nomad/command/helpers.go:455 +0x41f
github.com/hashicorp/nomad/command.(*JobRunCommand).Run(0xc00001f110, 0xc000136030, 0x2, 0x2, 0xc000453c40)
	github.com/hashicorp/nomad/command/job_run.go:197 +0x6ef
github.com/mitchellh/cli.(*CLI).Run(0xc00015cf00, 0xc00015cf00, 0xc0003ba240, 0x3b)
	github.com/mitchellh/cli@v1.1.0/cli.go:260 +0x1cf
main.RunCustom(0xc000136010, 0x4, 0x4, 0xc00010e058)
	github.com/hashicorp/nomad/main.go:142 +0x496
main.Run(...)
	github.com/hashicorp/nomad/main.go:87
main.main()
	github.com/hashicorp/nomad/main.go:83 +0x65
```
### Job file (if appropriate)
N/A

**Comment by jrasell** [comment]
Hi @andrewgwalker and apologies you ran into this. This issue has been fixed as part of the `1.0.6` and `1.1.0` releases via PR #10569, which I have confirmed locally. Please let me know if you have any follow up questions before I close this issue.

**Comment by andrewgwalker** [comment]
@jrasell That's great news! We'll update to the latest version, glad it got fixed.
